apps/admin_app/assets/js/components/CreatePromotions.tsx

```diff
@@ -0,0 +1,369 @@
+import * as React from "react";
+import { DatePicker, Input, Select, Button, InputNumber, Switch, Icon} from 'antd';
+import 'antd/dist/antd.css';
+import '../../css/promotions.css';
+import {Link, Redirect} from 'react-router-dom';
+
+export class PromotionsForm extends React.Component<{},any>{
+    constructor(props){
+        super(props);
+        this.state={
+            code:"",
+            name:"",
+            description:"",
+            starts_at:"",
+            expires_at:"",
+            usageLimit:"",
+            usageCount:"Nil",
+            matchPolicy:"",
+            activeStatus:false,
+            success:false,
+            ruleDown:false,
+            addRule:false,
+            actiondown:false,
+            addedRules:[],
+            availableRules:"",
+            addAction:false,
+            availableActions:"",
+            availableCalculators:"",
+            selectedRule:"",
+            selectedCalc:"",
+            lowRange:"",
+            upRange:"",
+            selectedProducts:"",
+            productMatchPolicy:"",
+            addedActions:[],
+            selectedAction:"",
+            selectedCalculator:""
+        }
+
+        this.handleStartDate = this.handleStartDate.bind(this);
```

Comment:
use arrow functions instead. doing bind inside constructor will slow down the initialization of this class

Reply:
@SakinaBoriwala21 the way you are using assignment and not using space is not the best way tbh. 

Avoid this. 👇 
```
this.state={
            code:"",
            name:"",
            description:"",
            starts_at:"",
```
Notice how I am leaving spaces in b/w the operators. This is largely the problem across the file. May I please request you to change it wherever necessary. 

A better way is to do it like this 👇 
```
this.state = {
            code: "",
            name: "",
            description: "",
            starts_at: "",
```

Let me know your thoughts if React community follows a different convention.